From 49d7969025f65d0c970c1f787bf9e273a3a7b6e5 Mon Sep 17 00:00:00 2001
From: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
Date: Fri, 12 Aug 2016 09:57:52 +0700
Subject: [PATCH] FDPM kernel module update version to 1.5.0

Currently, version of FDPM kernel module is 1.3.0 (source code from driver package of RZ)
While source code of FDPM kernel module from Yocto 1.10 of R-Car has version 1.5.0
This patch file update version of FDPM kernel module to 1.5.0 base on Yocto 1.10 of R-Car

Signed-off-by: Chien Nguyen <chien.nguyen.eb@rvc.renesas.com>
---
 drv/Makefile                | 5 ++++-
 drv/fdpm_sub.c              | 7 +++----
 drv/manager/fdpm_resource.c | 2 +-
 3 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/drv/Makefile b/drv/Makefile
index 00874a9..1672008 100755
--- a/drv/Makefile
+++ b/drv/Makefile
@@ -12,11 +12,14 @@ EXTRA_CFLAGS += $(U_INCLUDE)
 KERNELINC ?= $(KERNELDIR)/include/linux
 EXTRA_CFLAGS += -I$(KERNELINC)
 
+ifndef FDPM_MMNGRSYMVERS
+FDPM_MMNGRSYMVERS = Module.symvers
+endif
 
 all:
 	$(CP) ../include/fdpm_drv.h $(KERNELINC)
 	$(CP) ../include/fdpm_public.h $(KERNELINC)
 	$(CP) ../include/fdpm_api.h $(KERNELINC)
-	make -C $(KERNELDIR) M=$(PWD) KBUILD_EXTRA_SYMBOLS=$(FDPM_MMNGRDIR)/Module.symvers modules
+	make -C $(KERNELDIR) M=$(PWD) KBUILD_EXTRA_SYMBOLS=$(FDPM_MMNGRDIR)/$(FDPM_MMNGRSYMVERS) modules
 clean:
 	make -C $(KERNELDIR) M=$(PWD) clean
diff --git a/drv/fdpm_sub.c b/drv/fdpm_sub.c
index daceed5..e4652e2 100755
--- a/drv/fdpm_sub.c
+++ b/drv/fdpm_sub.c
@@ -141,7 +141,6 @@ int fdpm_thread(struct fdpm_privdata *priv)
 
 int fdpm_init(struct fdpm_privdata *priv)
 {
-	int                    ret;
 	int                    ret_val;
 	struct fdpm_drvdata    *pdrv = priv->pdrv;
 	struct platform_device *pdev = pdrv->pdev;
@@ -208,7 +207,7 @@ int fdpm_init(struct fdpm_privdata *priv)
 	if (ret_val != 0) {
 		APRINT("failed to fdpm_lib_driver_initialize %d (%ld)\n",
 		       ret_val, sub_ercd);
-		ret = -EFAULT;
+		ercd = -EFAULT;
 		goto exit;
 	}
 
@@ -216,7 +215,7 @@ int fdpm_init(struct fdpm_privdata *priv)
 	if (ret_val != 0) {
 		APRINT("failed to fdpm_lib_driver_initialize %d (%ld)\n",
 		       ret_val, sub_ercd);
-		ret = -EFAULT;
+		ercd = -EFAULT;
 		goto exit;
 	}
 
@@ -238,7 +237,7 @@ exit_clk1:
 exit_clk0_1:
 	clk_put(pdrv->fdpm_clk[0]);
 exit:
-	return ret;
+	return ercd;
 }
 
 int fdpm_lib_driver_initialize(struct platform_device *pdev,
diff --git a/drv/manager/fdpm_resource.c b/drv/manager/fdpm_resource.c
index 478f81f..8c79c08 100755
--- a/drv/manager/fdpm_resource.c
+++ b/drv/manager/fdpm_resource.c
@@ -473,7 +473,7 @@ int fdpm_resource_release(int ocmode, int vmode, int dev_no,
 		}
 	}
 
-	if (fdpm_pdata->fdpm_hw[dev_no].sumsize - size < 0)
+	if (fdpm_pdata->fdpm_hw[dev_no].sumsize < size)
 		fdpm_pdata->fdpm_hw[dev_no].sumsize = 0;
 	else
 		fdpm_pdata->fdpm_hw[dev_no].sumsize =
-- 
1.9.1

